engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugMetricsSystem.java

```diff
@@ -95,8 +98,9 @@ public MetricsMode toggle() {
      * @return True if the MetricsMode instance was in the set. False otherwise.
```

Comment:
Above, in the toggle() method, I'd add the following comment line, directed to fellow developers, between the end of the while loop and the return statement:

// This loop will always find at least one available MetricsMode instance, the defaultMode, 
// hence never running indefinitely. 

It is possible of course to arrive to that conclusion by looking at the rest of the code, but I figure we can make things easier to the techno-archeologist who will dig up this piece of code 1378 yeas from now. =)